features/developer_portal/admin/applications/plan_change.feature

```diff
@@ -0,0 +1,104 @@
+@javascript
+Feature: Developer portal change application plan
+
+  Background:
+    Given a provider
+    And the provider has "multiple_services" visible
+    And the provider has "service_plans" visible
+    And a product "The API"
+    And the following published application plans:
+      | Product | Name       | Cost per month | Default |
+      | The API | Developer  |                | true    |
+      | The API | Enterprise | 100            |         |
+    And a buyer "Jane" signed up to service "The API"
+    And the following application:
+      | Buyer | Name   | Plan      |
+      | Jane  | My App | Developer |
+    And the buyer logs in
+
+  Scenario: Change an application's plan requires approval
+    Given they go to the application's dev portal edit page
+    When follow "Review/Change"
+    And follow "Enterprise"
+    And press "Request Plan Change"
+    Then should see the flash message "A request to change your application plan has been sent"
+    And a message should be sent from buyer "Jane" to the provider with subject "API System: Plan change request" and body "Jane are requesting to have their plan changed to Enterprise"
+
+  Scenario: Change an application's plan directly
+    Given all the rolling updates features are off
+    And the product allows to change application plan directly
+    When they go to the application's dev portal edit page
+    And follow "Review/Change"
+    And follow "Enterprise"
+    And press "Change Plan"
+    Then should see the flash message "Plan change was successful."
+    And a message should be sent from buyer "Jane" to the provider with subject "API System: Application plan change" and body "plan from Developer to Enterprise"
+
+  Scenario: Change plan workflow dependent on credit card presence
+    Given all the rolling updates features are off
+    And the product allows to change application plan only with credit card
+    When they go to the application's dev portal edit page
+    And follow "Review/Change"
+    And follow "Enterprise"
+    And press "Request Plan Change"
+    Then they should see "A request to change your application plan has been sent."
+
+    Given the buyer has a valid credit card
```

Comment:
I was told this was bad... (multiple Given per scenario)

But OK :grimacing:

Reply:
Yes, this is bad, it's a result of copy-paste and I missed it.